### PR TITLE
[RFC] training plan planning ISO 9001

### PR DESCRIPTION
**Pull request by FernandoRomera**

### CODE REVIEW

**Comment by FernandoRomera** [comment]
@etobella Can you comment this RFC?

File: hr_course/models/hr_course.py [hunk 1]
```diff
@@ -5,57 +5,41 @@
...
         states={"draft": [("readonly", False)]},
-        tracking=True,
+        track_visibility="onchange",
     )
     end_date = fields.Date(
         string="End date",
         readonly=True,
         states={"draft": [("readonly", False)]},
-        tracking=True,
+        track_visibility="onchange",
```
Comment (etobella) [inline hr_course/models/hr_course.py]: On 13 you must use tracking:

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0

**Comment by etobella** [comment]
Would be great if you could add an explanation of the change, just to have some context

**Comment by etobella** [comment]
Sorry, I was missing the issue. I tested and worked fine. If you fix the comments I think it is great

**Comment by FernandoRomera** [comment]
@etobella would you have time for a review?

**Review by etobella — APPROVED** [review]
LGTM please squash your commits

File: hr_course/migrations/13.0.2.0.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_models(env, _models_renames)
+
+    openupgrade.logged_query(
+        env.cr,
+        """
+        SELECT id, name, category_id, permanence, permanence_time
+        INTO hr_course
```
Comment (etobella) [inline hr_course/migrations/13.0.2.0.0/pre-migration.py]: I would prefer to rewrite id in order to be sure that no we have no problem with the identifier sequence. On the other hand, this kinds of inserts are usually done on post-migration.

When you want to manage this kind of problems, you should need to add an old_course_id (or something like that) on hr.course.schedule in order to make it work (similar to account.invoice to account.move migration from 12 to 13)

Comment (FernandoRomera) [reply hr_course/migrations/13.0.2.0.0/pre-migration.py]: Can you review the solution?

**Review by AntoniRomera — APPROVED** [review]

**Review by olgamarcocb — APPROVED** [review]
LGTM :+1:

File: hr_course/migrations/13.0.2.0.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from openupgradelib import openupgrade  # pylint: disable=W7936
+
+_models_renames = [
+    ("hr.course", "hr.course.schedule"),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_models(env, _models_renames)
```
Comment (etobella) [inline hr_course/migrations/13.0.2.0.0/pre-migration.py]: openupgrade.rename_models(env.cr, _models_renames)

**Comment by etobella** [comment]
Migration script is not working. I will propose the migration script on a PR to your branch

**Comment by etobella** [comment]
https://github.com/FernandoRomera/hr/pull/1

This PR fixes migration scripts. Tested on our development environment successfully.

**Comment by etobella** [comment]
@FernandoRomera

**Comment by FernandoRomera** [comment]
@etobella Thanks.
Can you merge?

**Comment by etobella** [comment]
@OCA/human-resources-maintainers

**Review by etobella — APPROVED** [review]

**Comment by Saran440** [comment]
@FernandoRomera Could you squash your commits. please?

**Comment by FernandoRomera** [comment]
@Saran440 Thanks. Is ok for you?

**Comment by etobella** [comment]
Remove the merge branch (do a rebase if you need it). Also, change the commit description by `[REF] hr_course: training planning ISO 9001`

**Comment by FernandoRomera** [comment]
@etobella 
My experience with git is little and I've been trying all day, can you help me?

**Comment by etobella** [comment]
Sure!!! Can I take your Pr an propose a new one keeping your commits?

**Comment by FernandoRomera** [comment]
@etobella 
From my side there is no problem, I would be very grateful.

**Comment by pedrobaeza** [comment]
Superseded by #1050

**Pull request closed without merging**